symbols/overlay29.yml

```diff
@@ -214,6 +218,11 @@ overlay29:
         Returns true if you should get kicked out of the dungeon if an important team member (like the partner or certain story allies) faints.
         
         return: dungeon::nonstory_flag || dungeon::hidden_land_flag
+    - name: FadeToBlack
+      address:
+        EU: 0x22E12A8
+        NA: 0x22E0968
+      description: Fades the screen to black across several frames.
```

Comment:
Specify "No params." to make it obvious without needing to look at the headers. Saying nothing could mean that we don't know.

Reply:
Is it always so easy? I've certainly had experiences with more complicated functions where it's not always super obvious how many parameters there are. There's not always an explicit `mov` into registers before function calls if those registers are already populated beforehand by previous operations, and Ghidra sometimes infers the parameter list incorrectly. I've also quite often run into the conundrum of "is this function returning something, or does the end of the computation just happen to leave something in r0?" that can only be resolved by looking at the calling context.

I'm not sure what would be better as the "implicit default" when no parameter info is listed. My initial thought was to align ease of typing with ease of research: the less you know/the less work you've done, the less you need to type. That makes it easy to be lazy if you don't want to properly vet the parameter list of every single function while actively doing research, especially if you don't care that much about a particular function. It also means that if you forget to write "No params.", the meaning defaults to missing info, which isn't ideal but isn't _incorrect_. If it were the other way around, the meaning would default to no params, which could actually be wrong.

In practice, I do agree it's somewhat annoying to type "No params." We should optimize for the common, long-term case. And I guess my feeling is that the common case is going to be active research where there are a lot more unknowns and incomplete knowledge. I think your info dumps are somewhat of a special case.

Reply:
> Is it always so easy? I've certainly had experiences with more complicated functions where it's not always super obvious how many parameters there are.

The easiest way to check is usually to look at the context in which the function is called, since all the parameters have to be initialized there. Checking what the function does with the paramters is a good choice most of the time as well. The main problem comes if a function takes a parameter but doesn't use it or when a function takes paramters in the stack, since those stay there and aren't accessed until the function needs them.

> I've also quite often run into the conundrum of "is this function returning something, or does the end of the computation just happen to leave something in r0?"

Oh yeah, that one is harder to tell. You could have a function that returns something but in the current context the function that calls it doesn't use the return value for anything.

> We should optimize for the common, long-term case. And I guess my feeling is that the common case is going to be active research where there are a lot more unknowns and incomplete knowledge. I think your info dumps are somewhat of a special case.

Eh, also true. I don't have a strong preference for either option, we could leave it this way.

Reply:
> The easiest way to check is usually to look at the context in which the function is called, since all the parameters have to be initialized there. Checking what the function does with the paramters is a good choice most of the time as well. The main problem comes if a function takes a parameter but doesn't use it or when a function takes paramters in the stack, since those stay there and aren't accessed until the function needs them.

I'm glad there isn't an easier way that I'm just missing 😆. There are other problems I've run into related to optimizations that I alluded to. For example, if you have a chain of function calls (either nested of on the same level but right after one another), `r0` might get set at the top of the call stack and "pass through" to subsequent function calls, and if you're just looking at one of the intermediate functions, looking at the immediate calling code isn't going to help. The same thing happens when you have the return value of one function immediately passed into another function in r0.